src/common/dataHooks/useMediaSearch.js

```diff
@@ -31,34 +28,29 @@ function useMediaSearch({ type, library }) {
   const [searchTerm, setSearchTerm] = useState(searchParamsQuery)
   const [searchInputValue, setSearchInputValue] = useState(searchParamsQuery)
 
-  // Add search Term
-  const _searchParams = new URLSearchParams({
+  const siteSearchParams = new URLSearchParams({
     q: searchTerm,
     [TYPES]: type,
     [SORT]: searchTerm ? null : SORT_CREATED_DESC,
   })
 
-  const searchParamString = _searchParams.toString()
+  const sharedMediaSearchParams = new URLSearchParams({
+    q: searchTerm,
+    [TYPES]: type,
+    [SORT]: searchTerm ? null : SORT_CREATED_DESC,
+    [HAS_SITE_FEATURE]: SHARED_MEDIA,
+  })
 
   // Fetch search results
-  const response = useInfiniteQuery({
-    queryKey: [SEARCH, siteToSearch, searchParamString],
-    queryFn: ({ pageParam = 1 }) =>
-      api.search.get({
-        sitename: siteToSearch,
-        searchParams: searchParamString,
-        pageParam,
-      }),
-    getNextPageParam: (currentPage) => currentPage.next,
-    refetchOnWindowFocus: false,
-    refetchOnReconnect: false,
-    select: (responseData) => ({
-      pages: searchResponseAdaptor(responseData),
-      pageParams: responseData.pageParams,
-    }),
+  const siteSearchResponse = useSearchLoader({ searchParams: siteSearchParams })
+  const sharedMediaSearchResponse = useSearchAllSitesLoader({
+    searchParams: sharedMediaSearchParams,
+    enabled: searchSharedMedia,
```

Comment:
I can see where useSearchAllSitesLoader only makes a data request if it is enabled, but what is stopping useSearchLoader from making a data request every time we use this hook?

Reply:
Well spotted. Nothing is stopping it. I can talk you through my rationale if you like.